include/irr/asset/format/encodePixels.h

```diff
@@ -7,9 +7,43 @@
 #include "irr/core/core.h"
 #include "irr/asset/format/EFormat.h"
 
-
 namespace irr { namespace video
 {	
+
+	template<typename T>
+	inline uint64_t getRangeValueOfVariable(bool maxValue = true)
+	{
+		if (std::is_same<T, uint8_t>::value)
+			return 255ull * (maxValue ? 1 : 0);
+		else if (std::is_same<T, uint16_t>::value)
+			return 	65535ull * (maxValue ? 1 : 0);
+		else if (std::is_same<T, uint32_t>::value)
+			return 	4294967295ull * (maxValue ? 1 : 0);
+
+		else if (std::is_same<T, int8_t>::value)
+			return 127ull * (maxValue ? 1 : -1);
+		else if (std::is_same<T, int16_t>::value)
+			return 32767ull * (maxValue ? 1 : -1);
+		else if (std::is_same<T, int32_t>::value)
+			return 2147483647ull * (maxValue ? 1 : -1);
+		else
+			return -1; // handle an error
+	}
+
+	template<typename T, typename U>
+	inline void clampVariableProperly(T& variableToAssignClampingTo, const U& variableToClamp, bool isItNormalizedVariable = true)
```

Comment:
input is always a `double`

you can take away the boolean argument 
we have a function that is a template of `E_FORMAT` I believe its called `isFormatNormalized<E_FORMAT format>()` or something like that.